javascript/javascript3/week1/giphy/main.js

```diff
@@ -0,0 +1,62 @@
+const wordSearch = document.querySelector("#keyword");
+const resultsNum = document.querySelector("#resultsNum");
+const btn = document.querySelector("button");
+const results = document.querySelector(".results");
+const message = document.querySelector("#message");
+
+//just for me not to copy it all the time I'm modifying or refactoring functions
+const apiKey = "rebtDagnzLQZaDSW3TrUCB1mO0C7gja0";
+
+//declating a result array as a global scope as it will come up in different functions with modified values
+let resultsArray;
+
+//event listeners
+btn.addEventListener("click", searchingGiphs);
+resultsNum.addEventListener("input", sliceResults);
```

Comment:
Slicing your results like this works well in most cases. However, let me suggest using the `limit` parameter the giphy API supports in this case, because it will only load as many gifs as the user wishes to see and enable loading more than the default limit, reducing network traffic as well 🙂 

Another issue this event listeners has, is that it only slices the results when a user changes the number of results input value. If the value is already set and the search is changed, the number of results is not adjusted 🤓

Reply:
Thank you for the really detailed and helpful feedback. Fixed this point :)